.github/workflows/ci_cron_weekly.yml

```diff
@@ -1,47 +1,100 @@
+  #     with:
+  #       python-version: ${{ matrix.python }}
+  #   - name: Install language-pack-de and tzdata
+  #     if: ${{ matrix.os == 'ubuntu-latest' }}
+  #     run: sudo apt-get install language-pack-de tzdata
+  #   - name: Install graphviz
+  #     if: ${{ matrix.toxenv == 'linkcheck' }}
+  #     run: sudo apt-get install graphviz
+  #   - name: Install Python dependencies
+  #     run: python -m pip install --upgrade tox codecov
+  #   - name: Run tests
+  #     run: tox ${{ matrix.toxargs}} -e ${{ matrix.toxenv}} -- ${{ matrix.toxposargs}}
+
+
+  tests_more_architectures:
+
+    # The following architectures are emulated and are therefore slow, so
+    # we include them just in the weekly cron. These also serve as a test
+    # of using system libraries and using pytest directly.
+
+    runs-on: ubuntu-18.04
+    name: Python 3.7 on ${{ matrix.arch }}
+
     strategy:
       fail-fast: false
       matrix:
         include:
+          - arch: aarch64
+          - arch: ppc64le
+          - arch: s390x
 
-          - name: Documentation link check
-            os: ubuntu-latest
-            python: 3.7
-            toxenv: linkcheck
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: uraimo/run-on-arch-action@v2.0.5
+        name: Run tests
+        id: build
+        with:
+          arch: ${{ matrix.arch }}
+          distro: buster
 
-          # TODO: Uncomment when 3.10 is more mature. Should we use devdeps?
-          # Test against Python dev in cron job.
-          #- name: Python dev with basic dependencies
-          #  os: ubuntu-latest
-          #  python: 3.10-dev
-          #  toxenv: pydev-test
+          shell: /bin/sh
 
-    steps:
-    - name: Checkout code
-      uses: actions/checkout@v2
-      with:
-        fetch-depth: 0
-    - name: Set up Python
-      uses: actions/setup-python@v2
-      with:
-        python-version: ${{ matrix.python }}
-    - name: Install language-pack-de and tzdata
-      if: ${{ matrix.os == 'ubuntu-latest' }}
-      run: sudo apt-get install language-pack-de tzdata
-    - name: Install graphviz
-      if: ${{ matrix.toxenv == 'linkcheck' }}
-      run: sudo apt-get install graphviz
-    - name: Install Python dependencies
-      run: python -m pip install --upgrade tox codecov
-    - name: Run tests
-      run: tox ${{ matrix.toxargs}} -e ${{ matrix.toxenv}} -- ${{ matrix.toxposargs}}
+          install: |
+            apt-get update -q -y
+            apt-get install -q -y git \
+                      python3 \
+                      python3-configobj \
+                      python3-numpy \
+                      python3-ply \
+                      python3-venv \
+                      cython3 \
+                      wcslib-dev \
+                      libcfitsio-dev \
+                      liberfa1
+
+          run: |
+            python3 -m venv --system-site-packages tests
+            source tests/bin/activate
```

Comment:
This line fails for some reason!? https://github.com/astropy/astropy/pull/11045/checks?check_run_id=1402190795#step:3:853
I can reproduce this if I create a `.sh` file with the above commands with the `#!/bin/sh` - it gets executed via `sh` rather than bash.   Does one need to install `bash`? 

Though just replacing `source tests/bin/activate` with `. tests/bin/activate` does seem to make things work (though on my machine there are then errors collecting the tests).

```
Successfully tagged run-on-arch-astropy-astropy-weekly-cron-ppc64le-buster:latest
/home/runner/work/_actions/uraimo/run-on-arch-action/v2.0.5/src/run-on-arch-commands.sh: 4: /home/runner/work/_actions/uraimo/run-on-arch-action/v2.0.5/src/run-on-arch-commands.sh: source: not found
Error: The process '/home/runner/work/_actions/uraimo/run-on-arch-action/v2.0.5/src/run-on-arch.sh' failed with exit code 127
```

Reply:
Note that similar lines worked fine on regular architectures over at `pyerfa` - see https://github.com/liberfa/pyerfa/pull/58 - so if it is not clear what is going wrong here, perhaps we should simply revert to using `tox` and turn one of the other tests into one that uses the system libraries?